FlashCards.AhmadJer99/FlashCards/Managers/StacksManager.cs

```diff
@@ -0,0 +1,194 @@
+            string? readResult = Console.ReadLine();
+
+            if (string.IsNullOrEmpty(readResult))
+            {
+                AnsiConsole.MarkupLine("[red]Error- Invalid input[/]");
+                continue;
+            }
+            string userEntry = readResult.Trim();
+            exitMenu = userEntry.Equals("quit", StringComparison.CurrentCultureIgnoreCase);
+            if (exitMenu)
+                continue;
+
+            if (!StackNameExist(userEntry))
+            {
+                AnsiConsole.MarkupLine("[red]Error- Invalid input, please make sure you entered a correct stack name - Case Matters[/]");
+                continue;
+            }
+            var stackId = FetchModelId(userEntry);
+            AnsiConsole.MarkupLine($"[green]Current working stack = {userEntry}![/]");
+            return stackId;
+        }
+        while (!exitMenu);
+        return -1;
+    }
+
+    private bool StackNameExist(string userEntry)
+    {
+        foreach (StackDto stackDto in stacksDtos)
+        {
+            if (userEntry == stackDto.name)
+                return true;
+        }
+        return false;
+    }
+
+    private int FetchModelId(string stackName)
+    {
+        foreach (Stack stack in stacks)
+        {
+            if (stackName == stack.name)
+                return stack.id;
+        }
+        return -1;
+    }
+
+    protected override void DeleteModel(int stackId)
+    {
+        AnsiConsole.MarkupLine("[red]Are you sure you want to delete the stack?\n[white](To Confirm Deletion Pres Enter)[/][/]");
+        if (Console.ReadKey().Key == ConsoleKey.Enter)
+        {
+            stacksDBController.DeleteRow(stackId);
+            AnsiConsole.MarkupLine("[green]Stack Deleted Succesfully![/]");
+            return;
+        }
+        AnsiConsole.MarkupLine("[red]Stack Deletion Cancelled! (Press Any Key To Continue)[/]");
+        Console.ReadKey();
+    }
+
+    protected override void UpdateModel(int stackId, Stack modifiedStack)
+    {
+        modifiedStack.id = stackId;
+        try
+        {
+            stacksDBController.UpdateRow(modifiedStack);
+            AnsiConsole.MarkupLine("[green]Stack Updated Successfully![/]");
+        }
+        catch (SqlException ex)
+        {
+            if (!ex.Message.Contains("Violation of UNIQUE KEY constraint"))
+                throw;
+            AnsiConsole.MarkupLine("[red]Unique constraint violated - A Stack with that name already exists![/]");
+        }
+
+
+    }
+
+    protected override void AddNewModel()
+    {
+        var newStack = new Stack();
+
+        var stackName = AnsiConsole.Ask<string>("[yellow]Enter the stack name: [/]");
+        newStack.name = stackName;
+
+        try
+        {
+            stacksDBController.InsertRow(newStack);
+            AnsiConsole.MarkupLine("[green]Stack Added Successfully![/]\n (Press Any Key To Continue)");
+            Console.ReadKey();
+        }
+        catch (SqlException ex)
```

Comment:
🟠 **Dependency Inversion Principle**

💡 You have tightly coupled your UI and Database. What happens if in the future you want to switch out to Sqlite, Postgres, MySQL?

🔧 Think about if this can be refactored, do you really need to catch `SqlException`? Could this be handled elsewhere? Could you just catch `Exception`?

Reply:
I fixed that to be a normal exception and if it happens to have a unique constraint kind of exception in the message my code now handles that